<a href="https://colab.research.google.com/github/valdirdpg/curso-dnc/blob/master/TreinamentoAutonomo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
teste = pd.read_csv('/content/sample_data/california_housing_train.csv')

In [14]:
casa = teste.rename(columns = {
    'longitude':'LONG','latitude':'LAT',
    'housing_median_age':'MEDIA_IDADE',
    'total_rooms': 'TOTAL_QUARTO'
    })


In [15]:
casa.to_excel('casas.xlsx')

In [16]:
casas = pd.read_excel('/content/casas.xlsx')

In [17]:
casas

Unnamed: 0    LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  \
0               0 -114.31  34.19           15          5612            1283   
1               1 -114.47  34.40           19          7650            1901   
2               2 -114.56  33.69           17           720             174   
3               3 -114.57  33.64           14          1501             337   
4               4 -114.57  33.57           20          1454             326   
...           ...     ...    ...          ...           ...             ...   
16995       16995 -124.26  40.58           52          2217             394   
16996       16996 -124.27  40.69           36          2349             528   
16997       16997 -124.30  41.84           17          2677             531   
16998       16998 -124.30  41.80           19          2672             552   
16999       16999 -124.35  40.54           52          1820             300   

       population  households  median_income  median_house_value  
0            1015         472         1.4936               66900  
1            1129         463         1.8200               80100  
2             333         117         1.6509               85700  
3             515         226         3.1917               73400  
4             624         262         1.9250               65500  
...           ...         ...            ...                 ...  
16995         907         369         2.3571              111400  
16996        1194         465         2.5179               79000  
16997        1244         456         3.0313              103600  
16998        1298         478         1.9797               85800  
16999         806         270         3.0147               94600  

[17000 rows x 10 columns]

In [20]:
teste.index

RangeIndex(start=0, stop=17000, step=1)

In [21]:
casa.index

RangeIndex(start=0, stop=17000, step=1)

In [22]:
casa.head()

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
0 -114.31  34.19         15.0        5612.0          1283.0      1015.0   
1 -114.47  34.40         19.0        7650.0          1901.0      1129.0   
2 -114.56  33.69         17.0         720.0           174.0       333.0   
3 -114.57  33.64         14.0        1501.0           337.0       515.0   
4 -114.57  33.57         20.0        1454.0           326.0       624.0   

   households  median_income  median_house_value  
0       472.0         1.4936             66900.0  
1       463.0         1.8200             80100.0  
2       117.0         1.6509             85700.0  
3       226.0         3.1917             73400.0  
4       262.0         1.9250             65500.0

In [23]:
casa.tail(10)

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
16990 -124.22  41.73         28.0        3003.0           699.0      1530.0   
16991 -124.23  41.75         11.0        3159.0           616.0      1343.0   
16992 -124.23  40.81         52.0        1112.0           209.0       544.0   
16993 -124.23  40.54         52.0        2694.0           453.0      1152.0   
16994 -124.25  40.28         32.0        1430.0           419.0       434.0   
16995 -124.26  40.58         52.0        2217.0           394.0       907.0   
16996 -124.27  40.69         36.0        2349.0           528.0      1194.0   
16997 -124.30  41.84         17.0        2677.0           531.0      1244.0   
16998 -124.30  41.80         19.0        2672.0           552.0      1298.0   
16999 -124.35  40.54         52.0        1820.0           300.0       806.0   

       households  median_income  median_house_value  
16990       653.0         1.7038             78300.0  
16991       479.0         2.4805             73200.0  
16992       172.0         3.3462             50800.0  
16993       435.0         3.0806            106700.0  
16994       187.0         1.9417             76100.0  
16995       369.0         2.3571            111400.0  
16996       465.0         2.5179             79000.0  
16997       456.0         3.0313            103600.0  
16998       478.0         1.9797             85800.0  
16999       270.0         3.0147             94600.0

In [24]:
casa.isnull()

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
0      False  False        False         False           False       False   
1      False  False        False         False           False       False   
2      False  False        False         False           False       False   
3      False  False        False         False           False       False   
4      False  False        False         False           False       False   
...      ...    ...          ...           ...             ...         ...   
16995  False  False        False         False           False       False   
16996  False  False        False         False           False       False   
16997  False  False        False         False           False       False   
16998  False  False        False         False           False       False   
16999  False  False        False         False           False       False   

       households  median_income  median_house_value  
0           False          False               False  
1           False          False               False  
2           False          False               False  
3           False          False               False  
4           False          False               False  
...           ...            ...                 ...  
16995       False          False               False  
16996       False          False               False  
16997       False          False               False  
16998       False          False               False  
16999       False          False               False  

[17000 rows x 9 columns]

In [26]:
casa.notnull()

LONG   LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
0      True  True         True          True            True        True   
1      True  True         True          True            True        True   
2      True  True         True          True            True        True   
3      True  True         True          True            True        True   
4      True  True         True          True            True        True   
...     ...   ...          ...           ...             ...         ...   
16995  True  True         True          True            True        True   
16996  True  True         True          True            True        True   
16997  True  True         True          True            True        True   
16998  True  True         True          True            True        True   
16999  True  True         True          True            True        True   

       households  median_income  median_house_value  
0            True           True                True  
1            True           True                True  
2            True           True                True  
3            True           True                True  
4            True           True                True  
...           ...            ...                 ...  
16995        True           True                True  
16996        True           True                True  
16997        True           True                True  
16998        True           True                True  
16999        True           True                True  

[17000 rows x 9 columns]

In [27]:
casa.isnull().sum()

LONG                  0
LAT                   0
MEDIA_IDADE           0
TOTAL_QUARTO          0
total_bedrooms        0
population            0
households            0
median_income         0
median_house_value    0
dtype: int64

In [28]:
casa.notnull().sum()

LONG                  17000
LAT                   17000
MEDIA_IDADE           17000
TOTAL_QUARTO          17000
total_bedrooms        17000
population            17000
households            17000
median_income         17000
median_house_value    17000
dtype: int64

In [29]:
casa.filter(like='media')

median_income  median_house_value
0             1.4936             66900.0
1             1.8200             80100.0
2             1.6509             85700.0
3             3.1917             73400.0
4             1.9250             65500.0
...              ...                 ...
16995         2.3571            111400.0
16996         2.5179             79000.0
16997         3.0313            103600.0
16998         1.9797             85800.0
16999         3.0147             94600.0

[17000 rows x 2 columns]

In [32]:
casa[casa['LONG'].notnull()]

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
0     -114.31  34.19         15.0        5612.0          1283.0      1015.0   
1     -114.47  34.40         19.0        7650.0          1901.0      1129.0   
2     -114.56  33.69         17.0         720.0           174.0       333.0   
3     -114.57  33.64         14.0        1501.0           337.0       515.0   
4     -114.57  33.57         20.0        1454.0           326.0       624.0   
...       ...    ...          ...           ...             ...         ...   
16995 -124.26  40.58         52.0        2217.0           394.0       907.0   
16996 -124.27  40.69         36.0        2349.0           528.0      1194.0   
16997 -124.30  41.84         17.0        2677.0           531.0      1244.0   
16998 -124.30  41.80         19.0        2672.0           552.0      1298.0   
16999 -124.35  40.54         52.0        1820.0           300.0       806.0   

       households  median_income  median_house_value  
0           472.0         1.4936             66900.0  
1           463.0         1.8200             80100.0  
2           117.0         1.6509             85700.0  
3           226.0         3.1917             73400.0  
4           262.0         1.9250             65500.0  
...           ...            ...                 ...  
16995       369.0         2.3571            111400.0  
16996       465.0         2.5179             79000.0  
16997       456.0         3.0313            103600.0  
16998       478.0         1.9797             85800.0  
16999       270.0         3.0147             94600.0  

[17000 rows x 9 columns]

In [43]:
casa['LAT'].duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
16995     True
16996     True
16997    False
16998     True
16999     True
Name: LAT, Length: 17000, dtype: bool

In [44]:
casa.tail(20)

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
16980 -124.18  40.78         34.0        1592.0           364.0       950.0   
16981 -124.18  40.78         33.0        1076.0           222.0       656.0   
16982 -124.18  40.62         35.0         952.0           178.0       480.0   
16983 -124.19  41.78         15.0        3140.0           714.0      1645.0   
16984 -124.19  40.78         37.0        1371.0           319.0       640.0   
16985 -124.19  40.77         30.0        2975.0           634.0      1367.0   
16986 -124.19  40.73         21.0        5694.0          1056.0      2907.0   
16987 -124.21  41.77         17.0        3461.0           722.0      1947.0   
16988 -124.21  41.75         20.0        3810.0           787.0      1993.0   
16989 -124.21  40.75         32.0        1218.0           331.0       620.0   
16990 -124.22  41.73         28.0        3003.0           699.0      1530.0   
16991 -124.23  41.75         11.0        3159.0           616.0      1343.0   
16992 -124.23  40.81         52.0        1112.0           209.0       544.0   
16993 -124.23  40.54         52.0        2694.0           453.0      1152.0   
16994 -124.25  40.28         32.0        1430.0           419.0       434.0   
16995 -124.26  40.58         52.0        2217.0           394.0       907.0   
16996 -124.27  40.69         36.0        2349.0           528.0      1194.0   
16997 -124.30  41.84         17.0        2677.0           531.0      1244.0   
16998 -124.30  41.80         19.0        2672.0           552.0      1298.0   
16999 -124.35  40.54         52.0        1820.0           300.0       806.0   

       households  median_income  median_house_value  
16980       317.0         2.1607             67000.0  
16981       236.0         2.5096             72200.0  
16982       179.0         3.0536            107000.0  
16983       640.0         1.6654             74600.0  
16984       260.0         1.8242             70000.0  
16985       583.0         2.4420             69000.0  
16986       972.0         3.5363             90100.0  
16987       647.0         2.5795             68400.0  
16988       721.0         2.0074             66900.0  
16989       268.0         1.6528             58100.0  
16990       653.0         1.7038             78300.0  
16991       479.0         2.4805             73200.0  
16992       172.0         3.3462             50800.0  
16993       435.0         3.0806            106700.0  
16994       187.0         1.9417             76100.0  
16995       369.0         2.3571            111400.0  
16996       465.0         2.5179             79000.0  
16997       456.0         3.0313            103600.0  
16998       478.0         1.9797             85800.0  
16999       270.0         3.0147             94600.0

In [45]:
casa_duplicada = casa[casa.duplicated()]

In [46]:
casa_duplicada

Empty DataFrame
Columns: [LONG, LAT, MEDIA_IDADE, TOTAL_QUARTO, total_bedrooms, population, households, median_income, median_house_value]
Index: []

In [34]:
casa2 = casa[casa['LONG'].notna()]

In [35]:
casa2

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
0     -114.31  34.19         15.0        5612.0          1283.0      1015.0   
1     -114.47  34.40         19.0        7650.0          1901.0      1129.0   
2     -114.56  33.69         17.0         720.0           174.0       333.0   
3     -114.57  33.64         14.0        1501.0           337.0       515.0   
4     -114.57  33.57         20.0        1454.0           326.0       624.0   
...       ...    ...          ...           ...             ...         ...   
16995 -124.26  40.58         52.0        2217.0           394.0       907.0   
16996 -124.27  40.69         36.0        2349.0           528.0      1194.0   
16997 -124.30  41.84         17.0        2677.0           531.0      1244.0   
16998 -124.30  41.80         19.0        2672.0           552.0      1298.0   
16999 -124.35  40.54         52.0        1820.0           300.0       806.0   

       households  median_income  median_house_value  
0           472.0         1.4936             66900.0  
1           463.0         1.8200             80100.0  
2           117.0         1.6509             85700.0  
3           226.0         3.1917             73400.0  
4           262.0         1.9250             65500.0  
...           ...            ...                 ...  
16995       369.0         2.3571            111400.0  
16996       465.0         2.5179             79000.0  
16997       456.0         3.0313            103600.0  
16998       478.0         1.9797             85800.0  
16999       270.0         3.0147             94600.0  

[17000 rows x 9 columns]

In [77]:
import pandas as pd

# Criando um exemplo de DataFrame
data =pd.DataFrame({'nome': ['João', 'Maria', 'José', 'João', 'José'], 
        'idade': [25, 22, 30, 25, 30], 
        'cidade': ['São Paulo', 'Rio de Janeiro', 'Belo Horizonte', 'São Paulo', 'Belo Horizonte']}) 


In [78]:
dados = data.head()

In [79]:
dados

nome  idade          cidade
0   João     25       São Paulo
1  Maria     22  Rio de Janeiro
2   José     30  Belo Horizonte
3   João     25       São Paulo
4   José     30  Belo Horizonte

In [92]:
dados_order = dados.sort_values(by=['idade'])

In [93]:
dados_order.head()

nome  idade          cidade
1  Maria     22  Rio de Janeiro
0   João     25       São Paulo
3   João     25       São Paulo
2   José     30  Belo Horizonte
4   José     30  Belo Horizonte

In [94]:
dados_order['idade'].var()

12.3

In [95]:
sum(dados_order['idade'])

132

In [98]:
import math as m
m.sqrt(132)

11.489125293076057

In [99]:
dados_order['idade'].std()

3.5071355833500366

In [100]:
m.sqrt(12.3)

3.5071355833500366

In [84]:
dados_order.head()

nome  idade          cidade
0   João     25       São Paulo
1  Maria     22  Rio de Janeiro
2   José     30  Belo Horizonte
3   João     25       São Paulo
4   José     30  Belo Horizonte

In [80]:
dados['idade'].mean()

26.4

In [81]:
dados['idade'].median()

25.0

In [57]:
data2 = data[data.duplicated()]

In [64]:
data.drop_duplicates(subset='nome', keep='first', inplace=True)

In [68]:
dados.head()

nome  idade          cidade
0   João     25       São Paulo
1  Maria     22  Rio de Janeiro
2   José     30  Belo Horizonte
3   João     25       São Paulo
4   José     30  Belo Horizonte

In [67]:
data.head()

nome  idade          cidade
0   João     25       São Paulo
1  Maria     22  Rio de Janeiro
2   José     30  Belo Horizonte

In [65]:
casa2.drop_duplicates(subset='LAT', keep='first', inplace=True)

In [66]:
casa2

LONG    LAT  MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  population  \
0     -114.31  34.19         15.0        5612.0          1283.0      1015.0   
1     -114.47  34.40         19.0        7650.0          1901.0      1129.0   
2     -114.56  33.69         17.0         720.0           174.0       333.0   
3     -114.57  33.64         14.0        1501.0           337.0       515.0   
4     -114.57  33.57         20.0        1454.0           326.0       624.0   
...       ...    ...          ...           ...             ...         ...   
16924 -124.10  40.73         33.0         644.0           129.0       334.0   
16958 -124.16  41.02         23.0        1672.0           385.0      1060.0   
16987 -124.21  41.77         17.0        3461.0           722.0      1947.0   
16988 -124.21  41.75         20.0        3810.0           787.0      1993.0   
16997 -124.30  41.84         17.0        2677.0           531.0      1244.0   

       households  median_income  median_house_value  
0           472.0         1.4936             66900.0  
1           463.0         1.8200             80100.0  
2           117.0         1.6509             85700.0  
3           226.0         3.1917             73400.0  
4           262.0         1.9250             65500.0  
...           ...            ...                 ...  
16924       121.0         3.9659            111800.0  
16958       390.0         2.1726             75500.0  
16987       647.0         2.5795             68400.0  
16988       721.0         2.0074             66900.0  
16997       456.0         3.0313            103600.0  

[840 rows x 9 columns]

In [69]:
casa3 = casa2.groupby(['MEDIA_IDADE']).mean().reindex()

In [70]:
casa3

LONG        LAT  TOTAL_QUARTO  total_bedrooms   population  \
MEDIA_IDADE                                                                     
2.0         -116.725000  33.420000   1253.500000      186.000000   366.500000   
3.0         -118.194000  35.378000   9874.800000     1675.800000  3798.800000   
4.0         -118.469412  35.201176   4647.470588      794.176471  2108.941176   
5.0         -117.999333  34.928667   4351.333333      830.266667  2306.000000   
6.0         -118.320909  35.209091   3309.363636      597.636364  1513.090909   
7.0         -119.135455  35.755455   6368.818182     1052.181818  2708.454545   
8.0         -119.687647  37.250588   2462.705882      458.647059  1432.588235   
9.0         -119.220833  36.992500   4240.500000      800.416667  1567.083333   
10.0        -119.490417  37.181667   3061.000000      632.291667  1243.000000   
11.0        -119.333889  36.983889   3852.611111      756.500000  1556.611111   
12.0        -119.426190  36.790476   3173.666667      571.095238  1461.095238   
13.0        -119.621290  36.966129   3237.129032      615.193548  1425.290323   
14.0        -119.691750  37.416250   3407.625000      625.200000  1192.400000   
15.0        -119.032963  36.910185   3423.592593      694.888889  1455.388889   
16.0        -119.485714  36.870238   2961.785714      575.238095  1270.642857   
17.0        -119.643231  37.495538   2264.630769      451.369231   992.307692   
18.0        -119.380816  36.933878   2479.918367      494.877551  1188.367347   
19.0        -119.233913  37.049783   2309.391304      454.478261  1074.673913   
20.0        -119.150244  36.827805   2343.341463      485.097561  1111.097561   
21.0        -118.809062  36.551250   2051.531250      452.906250   976.500000   
22.0        -120.210870  38.168696   2650.434783      555.565217  1202.695652   
23.0        -119.703750  37.249583   2189.208333      449.500000  1173.291667   
24.0        -119.129474  36.787368   1997.894737      448.000000  1159.526316   
25.0        -119.846500  37.458000   1731.350000      385.300000   982.150000   
26.0        -120.058000  37.929200   1983.560000      407.880000   936.520000   
27.0        -119.280625  36.613125   2306.562500      424.437500   967.062500   
28.0        -119.677333  37.208667   2362.933333      489.666667  1290.933333   
29.0        -118.486667  35.865333   2007.400000      418.466667  1049.133333   
30.0        -119.584615  37.133846   1939.076923      422.846154   866.615385   
31.0        -119.794000  37.361000   2020.500000      461.700000  1425.400000   
32.0        -118.993571  36.568571   1506.071429      357.642857  1027.142857   
33.0        -119.455000  37.235500   1869.700000      378.500000   927.200000   
34.0        -119.383077  37.346923   1993.000000      464.076923   728.384615   
35.0        -119.854000  37.507000   3633.000000      725.900000  2384.100000   
36.0        -119.723571  36.502143   1464.357143      310.571429   824.857143   
37.0        -118.636667  35.326667   1005.166667      221.833333   643.166667   
38.0        -118.742000  36.158000    849.400000      189.800000   668.600000   
39.0        -120.525000  37.502500   2846.000000      614.250000  1920.000000   
40.0        -119.020000  35.420000   1089.000000      226.000000   520.000000   
41.0        -114.590000  34.830000    812.000000      168.000000   375.000000   
42.0        -121.160000  41.780000   2918.000000      576.000000  1182.000000   
43.0        -120.190000  38.070000    102.000000       19.000000    44.000000   
44.0        -120.395000  37.130000   3117.000000      631.000000   887.000000   
45.0        -120.943333  37.920000   1656.000000      334.000000   834.000000   
46.0        -120.690000  37.770000    431.000000       86.000000   239.000000   
47.0        -124.070000  40.870000   1765.000000      326.000000   796.000000   
48.0        -114.610000  34.840000   1291.000000      248.000000   580.000000   
49.0        -121.230000  37.870000     98.000000  

In [71]:
casa4 = casa2.groupby(['total_bedrooms']).mean().reindex()

In [72]:
casa4

LONG     LAT  MEDIA_IDADE  TOTAL_QUARTO  population  \
total_bedrooms                                                          
2.0            -117.79  35.210          4.0           2.0         6.0   
3.0            -117.76  35.220          4.0          18.0         8.0   
7.0            -118.38  35.190         26.0          52.5        33.5   
18.0           -118.06  35.585          9.0         102.0        42.0   
19.0           -120.19  38.070         43.0         102.0        44.0   
...                ...     ...          ...           ...         ...   
2782.0         -122.17  38.080          7.0       18392.0      8276.0   
2837.0         -116.42  33.760         14.0       16921.0      2524.0   
2888.0         -117.04  33.150         15.0       13814.0      6583.0   
2953.0         -118.15  34.670          5.0       12317.0      6291.0   
3366.0         -117.19  33.900          3.0       21060.0      9623.0   

                households  median_income  median_house_value  
total_bedrooms                                                 
2.0                    2.0         2.3750            137500.0  
3.0                    6.0         1.6250            275000.0  
7.0                    8.0         3.6250            331250.5  
18.0                  19.0         4.8562             73750.0  
19.0                  13.0         0.4999            162500.0  
...                    ...            ...                 ...  
2782.0              2742.0         6.4232            229200.0  
2837.0              1262.0         7.6281            341700.0  
2888.0              2789.0         2.8247            150000.0  
2953.0              2654.0         3.5732            146900.0  
3366.0              2812.0         4.1890            143000.0  

[576 rows x 8 columns]

In [73]:
casa.describe()

LONG           LAT   MEDIA_IDADE  TOTAL_QUARTO  total_bedrooms  \
count  17000.000000  17000.000000  17000.000000  17000.000000    17000.000000   
mean    -119.562108     35.625225     28.589353   2643.664412      539.410824   
std        2.005166      2.137340     12.586937   2179.947071      421.499452   
min     -124.350000     32.540000      1.000000      2.000000        1.000000   
25%     -121.790000     33.930000     18.000000   1462.000000      297.000000   
50%     -118.490000     34.250000     29.000000   2127.000000      434.000000   
75%     -118.000000     37.720000     37.000000   3151.250000      648.250000   
max     -114.310000     41.950000     52.000000  37937.000000     6445.000000   

         population    households  median_income  median_house_value  
count  17000.000000  17000.000000   17000.000000        17000.000000  
mean    1429.573941    501.221941       3.883578       207300.912353  
std     1147.852959    384.520841       1.908157       115983.764387  
min        3.000000      1.000000       0.499900        14999.000000  
25%      790.000000    282.000000       2.566375       119400.000000  
50%     1167.000000    409.000000       3.544600       180400.000000  
75%     1721.000000    605.250000       4.767000       265000.000000  
max    35682.000000   6082.000000      15.000100       500001.000000

# SOBRE DROP_DUPLICATE

def drop_duplicates(subset: (Hashable | Sequence[Hashable] | None)=Nenhum, mantenha: (Literal['first'] | Literal['last'] | Literal[False])='first', inplace: bool=False , ignore_index: bool=False) ->(DataFrame | Nenhum)
Retorne DataFrame com linhas duplicadas removidas.

A consideração de determinadas colunas é opcional. Índices, incluindo índices de tempo
são ignorados.

Parâmetros

subconjunto: rótulo da coluna ou sequência de rótulos, opcional
    Considere apenas algumas colunas para identificar duplicatas, por
padrão usa todas as colunas.
manter: {'primeiro', 'último', False}, padrão 'primeiro'
    Determina quais duplicatas (se houver) manter.


first : Elimine as duplicatas, exceto a primeira ocorrência.

last : Solte as duplicatas, exceto a última ocorrência.

Falso: Elimine todas as duplicatas.
inplace: bool, padrão False
Se deve colocar duplicatas no lugar ou devolver uma cópia.
ignore_index : bool, padrão Falso
Se True, o eixo resultante será rotulado como 0, 1, …, n - 1.


devoluções

DataFrame ou nenhum
    DataFrame com duplicatas removidas ou None se inplace=True.

Veja também

DataFrame.value_counts: conta combinações exclusivas de colunas.

Exemplos

Considere o conjunto de dados contendo classificação de ramen.

>>> df = pd.DataFrame({
... 'marca': ['Yum Yum', 'Yum Yum', 'Indomie', 'Indomie', 'Indomie'],
... 'estilo': ['xícara', 'xícara', 'xícara', 'pacote', 'pacote'],
... 'avaliação': [4, 4, 3,5, 15, 5]
... })
>>> df
    classificação de estilo de marca
0 copo Yum Yum 4.0
1 copo Yum Yum 4.0
2 Copa Indomie 3.5
3 Indomie pack 15.0
4 Indomie pack 5.0
Por padrão, ele remove linhas duplicadas com base em todas as colunas.

>>> df.drop_duplicates()
    classificação de estilo de marca
0 copo Yum Yum 4.0
2 Copa Indomie 3.5
3 Indomie pack 15.0
4 Indomie pack 5.0
Para remover duplicatas em colunas específicas, use subconjunto.

>>> df.drop_duplicates(subset=['marca'])
    classificação de estilo de marca
0 copo Yum Yum 4.0
2 Copa Indomie 3.5
Para remover duplicatas e manter as últimas ocorrências, use keep.

>>> df.drop_duplicates(subset=['marca', 'estilo'], manter='último')
    classificação de estilo de marca
1 copo Yum Yum 4.0
2 Copa Indomie 3.5
4 Indomie pack 5.0